forem/forem
# Remove name from invitation flow

## What type of PR is this? (check all applicable)


## Description

This PR removes "name" from the admin invite user flow, so that now only an email address is requested/required. Instead, a user is prompted to enter their name when they accept the invitation.

In the previous admin flow, the new username was created based on the invitee's name, so instead we are now creating it based on the email address (taking everything to the left of the "@" sign, removing special characters, and appending a random number for uniqueness).

Since we no longer have a name to display, I've also removed the name from the invitations index (`/admin/invitations`), and the invitation email itself.

## Related Tickets & Documents



- Closes https://github.com/forem/forem/issues/16226

## QA Instructions, Screenshots, Recordings

- As an admin, visit `/admin/invitations`
- Click "invite member"
- Only an email address should be requested in the form
- Enter an email address and submit
- You should return to the invitations index page, and see the new user in the list of invites
- The username for the new invitee should be the alphanumeric characters from the first half of the email address, plus a random number


https://user-images.githubusercontent.com/20773163/165352280-d649239a-7ccf-414f-9368-3a26272fea25.mp4


- Check your terminal for a printout of the email and grab the invitation link. You will need to remove the `3D` at the beginning of the token (see Dan's comments below)
- Visit the link, and try to submit without entering a name. Check you see a validation error.
- Fill out all fields and submit again. You should see the onboarding flow reflecting your name.


https://user-images.githubusercontent.com/20773163/165937348-681d22ef-1b6b-474f-906e-9c52b8bc0d52.mp4



### UI accessibility concerns?

N/A

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

The linked issue is tagged for a changelog rollup

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![we have no names, we are nameless](https://media.giphy.com/media/P5wPrhzZDdeJW/giphy.gif)
